packages/esm-patient-registration-app/src/config-schema.ts

```diff
@@ -0,0 +1,173 @@
+  fieldConfigurations: {
+    name: {
+      displayMiddleName: boolean;
+      unidentifiedPatient: boolean;
+      defaultUnknownGivenName: string;
+      defaultUnknownFamilyName: string;
+    };
+  };
+  links: {
+    submitButton: string;
+  };
+  concepts: {
+    patientPhotoUuid: string;
+  };
+  defaultPatientIdentifierTypes: Array<string>;
+}
+
+export const builtInSections: Array<SectionDefinition> = [
+  {
+    id: 'demographics',
+    name: 'Basic Info',
+    fields: ['name', 'gender', 'dob', 'id'],
+  },
+  { id: 'contact', name: 'Contact Details', fields: ['address', 'phone & email'] },
+  { id: 'death', name: 'Death Info', fields: [] },
+  { id: 'relationships', name: 'Relationships', fields: [] },
+];
+
+export const builtInFields = ['name', 'gender', 'dob', 'address', 'id', 'phone & email'] as const;
+
+export const esmPatientRegistrationSchema = {
+  sections: {
+    _type: Type.Array,
+    _default: ['demographics', 'contact', 'relationships'],
+    _description: `An array of strings which are the keys from 'sectionDefinitions' or any of the following built-in sections: '${builtInSections
+      .map((s) => s.id)
+      .join("', '")}'.`,
+    _elements: {
+      _type: Type.String,
+    },
+  },
+  sectionDefinitions: {
+    _type: Type.Array,
+    _elements: {
+      id: {
+        _type: Type.String,
+        _description: `How this section will be referred to in the \`sections\` configuration. To override a built-in section, use that section's id. The built in section ids are '${builtInSections
+          .map((s) => s.id)
+          .join("', '")}'.`,
+      },
+      name: {
+        _type: Type.String,
+        _description: 'The title to display at the top of the section.',
+      },
+      fields: {
+        _type: Type.Array,
+        _default: [],
+        _description: `The parts to include in the section. Can be any of the following built-in fields: ${builtInFields.join(
+          ', ',
+        )}. Can also be an id from an object in the \`fieldDefinitions\` array, which you can use to define custom fields.`,
+        _elements: { _type: Type.String },
+      },
+    },
+    _default: [],
+  },
+  fieldDefinitions: {
+    _type: Type.Array,
+    _elements: {
+      id: {
+        _type: Type.String,
+        _description:
+          'How this field will be referred to in the `fields` element of the `sectionDefinitions` configuration.',
+      },
+      uuid: {
+        _type: Type.UUID,
+        _description: "Person attribute type UUID that this field's data should be saved to.",
+      },
+      label: {
+        _type: Type.String,
+        _default: null,
+        _description: 'The label of the input. By default, uses the metadata `display` attribute.',
+      },
+      placeholder: {
+        _type: Type.String,
+        _default: '',
+        _description: 'Placeholder that will appear in the input.',
+      },
+      validation: {
+        required: { _type: Type.Boolean, _default: false },
```

Comment:
Maybe doesn't need to be dealt with in this PR, but it would be nice to have this handled at some point.

Reply:
Yeah, I think it's just not used at present? That should be fixed.